src/tekton/save.ts

```diff
@@ -0,0 +1,33 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { pipelineExplorer } from '../pipeline/pipelineExplorer';
+import { getStderrString, Command } from '../tkn';
+import { cli } from '../cli';
+import { tektonYaml } from '../yaml-support/tkn-yaml';
+import { contextGlobalState } from '../extension';
+
+export async function save(document: vscode.TextDocument): Promise<void> {
+  let value: string;
+  if (document.uri.authority !== 'loadtektonresource') {
+    const verifyTknYaml = tektonYaml.isTektonYaml(document);
+    if (!contextGlobalState.workspaceState.get(document.uri.fsPath)) {
+      value = await vscode.window.showWarningMessage('Detected Tekton yaml resource file. Do you want to deploy to cluster?', 'Save', 'Save Once', 'Cancel');
```

Comment:
Can we change the message as 
`Detected Tekton resources. Do you want to deploy to cluster?`
and instead of Save can we use Deploy on the buttons So that they are `Deploy`, `Deploy Once`, `Cancel`